st2common/st2common/models/db/__init__.py

```diff
@@ -121,12 +129,15 @@ def aggregate(self, *args, **kwargs):
         return self.model.objects(**kwargs)._collection.aggregate(*args, **kwargs)
 
     @staticmethod
+    def add(instance):
+        instance.save(force_insert=True)
```

Comment:
This is to ensure mongodb document is inserted and not updated. The idea here is that if the doc exists then the lock is in use.  We don't want an accidental update here.